### PR TITLE
Use bs58 strings to declare IDs rather then raw bytes

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Using raw pubkey bytes to declare ids and programs is unnecessary and not as clear as other forms

#### Summary of Changes

Use bs58 strings

Fixes #

### CODE REVIEW

**Comment by garious** [comment]
Cool. What happens when the string is too short or too long?

**Comment by jackcmay** [comment]
@garious The macros include their own tests (like before) that will panic if the string cannot be decoded.
For example:
```
---- sysvar::clock::test_id stdout ----
thread 'sysvar::clock::test_id' panicked at 'called `Result::unwrap()` on an `Err` value: WrongSize', src/libcore/result.rs:999:5
```

**Comment by garious** [comment]
Okay. If not too much trouble, it'd be helpful to give a more informative error message.  I've hit that more than a few times, and wished it told me I had one too many or one too few characters.  Don't want to count all those characters. Don't want to guess and check.

**Comment by rob-solana** [comment]
> Okay. If not too much trouble, it'd be helpful to give a more informative error message. I've hit that more than a few times, and wished it told me I had one too many or one too few characters. Don't want to count all those characters. Don't want to guess and check.

would also like it if bs58 told me which character wasn't kosher

File: sdk/src/pubkey.rs [hunk 1]
```diff
@@ -104,40 +104,38 @@ pub fn read_pubkey_file(infile: &str) -> Result<Pubkey, Box<dyn error::Error>> {
...
+    ($bs58_string:expr) => (
+        use std::str::FromStr;
+
+        $crate::lazy_static::lazy_static! {
+            static ref _PUBKEY: $crate::pubkey::Pubkey = { $crate::pubkey::Pubkey::from_str(&$bs58_string).unwrap() };
+        }
 
         pub fn check_id(id: &$crate::pubkey::Pubkey) -> bool {
-            id.as_ref() == $id
+            *id == *_PUBKEY
```
Comment (rob-solana) [inline sdk/src/pubkey.rs]: id == _PUBKEY faster?

File: sdk/src/pubkey.rs [hunk 2]
```diff
@@ -104,40 +104,38 @@ pub fn read_pubkey_file(infile: &str) -> Result<Pubkey, Box<dyn error::Error>> {
...
         }
 
         pub fn id() -> $crate::pubkey::Pubkey {
-            $crate::pubkey::Pubkey::new(&$id)
+            (*_PUBKEY).clone()
```
Comment (rob-solana) [inline sdk/src/pubkey.rs]: pubkey does Copy, I think

**Comment by jackcmay** [comment]
@rob-solana How abut thiis:
`thread 'sysvar::clock::test_id' panicked at 'Error: SysvarC1ock1111l111111111111111111111111111, provided string contained invalid character 'l' at byte 15', sdk/src/sysvar/clock.rs:11:1
`
And
`thread 'sysvar::clock::test_id' panicked at 'Error: SysvarC1ock111111111111111111111111111111111, decoded length 33, expected 32', sdk/src/sysvar/clock.rs:11:1`

File: sdk/src/pubkey.rs (hunk 1 above)
Comment (jackcmay) [reply sdk/src/pubkey.rs]: I benched, this solution is the same speed as the original

**Comment by garious** [comment]
:shipit:

**Pull request merged**